spec/requests/api/graphql/queries/consignments_spec.rb

```diff
@@ -0,0 +1,278 @@
+      end
+    end
+
+    context 'with a request from a regular user' do
+      let(:token) do
+        payload = { aud: 'gravity', sub: 'userid', roles: 'user' }
+        JWT.encode(payload, Convection.config.jwt_secret)
+      end
+
+      it 'returns an error for that request' do
+        post '/api/graphql', params: { query: query }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        consignments_response = body['data']['consignments']
+        expect(consignments_response).to eq nil
+
+        error_message = body['errors'][0]['message']
+
+        expect(error_message).to eq "Can't find partner."
+      end
+    end
+  end
+
+  describe 'valid requests' do
+    context 'with a request from a partner' do
+      it 'returns the expected graphql response' do
+        post '/api/graphql', params: { query: query }, headers: headers
+        
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        expect(body['data']).to eq(
+          {
+            consignments: {
+              totalCount: 2,
+              edges: [
+                {
+                  node: {
+                    currency: "USD",
+                    state: "SOLD",
+                  }
+                },
+                {
+                  node: {
+                    currency: "USD",
+                    state: "BOUGHT_IN",
+                  }
+                }
+              ]
+            }
+          }.deep_stringify_keys
+        )
+
+
+      end
+      it 'returns only returns sold or bought in consignments' do
+        post '/api/graphql', params: { query: query }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        total_count = body.dig('data', 'consignments', 'totalCount')
+        state1 = body.dig('data', 'consignments', 'edges', 0, 'node', 'state')
+        state2 = body.dig('data', 'consignments', 'edges', 1, 'node', 'state')
+
+        expect(total_count).to eq 2
+
+        expect(state1). not_to eq 'OPEN'
+        expect(state1). not_to eq 'CANCELLED'
+        expect(state1). to eq 'SOLD'
+
+        expect(state2). not_to eq 'OPEN'
+        expect(state2). not_to eq 'CANCELLED'
+        expect(state2). to eq 'BOUGHT_IN'
+      end
+    end
+
+    context 'when asking for only sent consignments' do
+      let(:consigned_partner_submission) do
+        Fabricate :partner_submission, partner: partner
+      end
+      let!(:consigned_offer) do
+        Fabricate :offer,
+                  partner_submission: consigned_partner_submission,
+                  state: 'consigned'
+      end
+
+      let(:query_inputs) do
+        "gravityPartnerId: \"#{
+          partner.gravity_partner_id
+        }\""
+      end
+
+      it 'returns only the requested consignments' do
+        post '/api/graphql', params: { query: query }, headers: headers
```

Comment:
**question [potentially blocking]:**

I don't think I totally understand this test. [The `context` name suggests we're looking for only "sent" consignments], and [the setup indicates that we want a consignment in the state of `consigned`](). The query specifies we are looking to limit results by `gravityPartnerId` though, not by "sent", though. And the test asserts that there are 2 results, but it doesn't tell me how those results are limited to "sent" results only.

Can you help me understand the goal of this test? There seems to be a mismatch between what it says it's going to do and what it does, or I have a misunderstanding about what it says it's going to do.


_Comment format based on [conventionalcomments.org](https://conventionalcomments.org/)_

Reply:
Ah, these are leftover that should have been deleted! 👎  Just cleaned these up!